pkg/storage/s3/s3.go

```diff
@@ -211,35 +287,67 @@ func (d *driver) CreateStorage(cr *opapi.ImageRegistry, modified *bool) error {
 		return err
 	}
 
-	for i := 0; i < 5000; i++ {
-		if len(d.Config.Bucket) == 0 {
-			d.Config.Bucket = fmt.Sprintf("%s-%s-%s-%s", clusterconfig.StoragePrefix, d.Config.Region, strings.Replace(string(cv.Spec.ClusterID), "-", "", -1), strings.Replace(string(uuid.NewUUID()), "-", "", -1))[0:62]
+	// If a bucket name is supplied, and it already exists and we can access it
+	// just update the config
+	bucketExists, err := d.bucketExists(cr.Spec.Storage.S3.Bucket)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeNoSuchBucket, "Forbidden", "NotFound":
+				// If the bucket doesn't exist that's ok, we'll try to create it
+				util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionFalse, aerr.Code(), aerr.Error(), modified)
+			default:
+				util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionUnknown, "Unknown Error Occurred", err.Error(), modified)
+				return err
+			}
 		}
+	}
+	if len(cr.Spec.Storage.S3.Bucket) != 0 && bucketExists {
+		d.Config.Bucket = cr.Spec.Storage.S3.Bucket
+		cr.Status.Storage.State.S3 = d.Config
```

Comment:
How did we end up with "Status.Storage.State.S3" instead of just "Status.Storage.S3" to match the spec struct?

Reply:
We have Status.Storage.Managed, which isn't part of the configuration (but it might make sense for it to be, so I can refactor that)

Reply:
Updated in current push